price-service/src/priceMonitor.ts

```diff
@@ -10,71 +10,66 @@ export class PriceMonitor {
     private provider:HttpProvider;
     private iconService: IconService;
     private wallet: Wallet;
+    private address: string;
+    private nid: string;
 
     constructor(config: any) {
-        this.threshold = config.threshold / 100;
+        this.threshold = config.priceChangeThreshold / 100;
         this.minInterval = config.interval;
         this.provider = new HttpProvider(config.icon_url);
         this.iconService = new IconService(this.provider);
         this.wallet = Wallet.loadPrivateKey(config.icon_pk);
+        this.address = config.address;
+        this.nid = config.nid;
     }
 
     public async onPriceUpdate(feed: PriceFeed): Promise<void> {
-
         if (this.processing) {
-            console.log(`Skipping update`);
             return;
         }
 
         this.processing = true
 
         try {
-            const parsed = feed.parsed;
-
-            for (let priceEntry of parsed) {
+            for (let priceEntry of feed.parsed) {
                 let lastUpdate = this.prices.get(priceEntry.id);
-                if (lastUpdate) {
 
+                if (lastUpdate) {
                     const priceChange = Math.abs(priceEntry.price.price - lastUpdate.price) / lastUpdate.price;
-                    console.log(priceChange);
-
-                    if (priceChange >= this.threshold || priceEntry.price.publish_time - lastUpdate.publish_time >= this.minInterval) {
-                        await this.updatePrice(feed);
-                        return;
+                    const timeSinceLastUpdate = priceEntry.price.publish_time - lastUpdate.publish_time
+                    if (priceChange < this.threshold && timeSinceLastUpdate < this.minInterval) {
+                        continue;
                     }
-                } else {
-                    await this.updatePrice(feed);
-                    return;
                 }
 
+                await this.updatePrice(feed);
+                feed.parsed.forEach(priceEntry => {
+                    this.prices.set(priceEntry.id, priceEntry.price)
+                })
                 return;
```

Comment:
Yeah, since new streams always contains all pairs, and i get proof for all the configured once at the same time. Its was extra effort and rpc calls to split the proof of a separate pair.  However might be slightly more expensive and give more detailed prices than we would need